### ISSUE TITLE
What's the drift on the atmega clock? Is an RTC really necessary? 

### ISSUE THREAD

**Issue by dwblair** [p-v-o-s/riffle-ito#16]

**Comment by bgamari** [comment]
If I understand your point here the problem isn't really drift so much as it is power consumption.

**Comment by dwblair** [comment]
Note to self: read up, here: http://jeelabs.org/2010/10/18/tracking-time-in-your-sleep/

**Comment by dwblair** [comment]
Ah, hah -- yeah, I just started realizing that the issue (I think) is that if we put the 328 chip to sleep using the JeeLib 'watchdog timer' technique, for example, we're not quite sure what time it is when we wake up -- and that's where the external RTC comes in handy, I guess?  And then we could achieve even deeper sleep modes if we use an interrupt from the RTC to wake up the 328?  But in any case, maybe I was a bit rash in selecting the DS3231 as the RTC, when a lower-cost RTC would've done?

**Comment by walkerjeffd** [comment]
my understanding, which may be off, is that the RTC is handy for keeping track of the actual date/time thanks to the coin cell battery. the atmega can keep track of time, but doesn't know where in "time" it is. so if you lose power, then it doesn't know how long it slept for and thus doesn't know what time it is when it wakes up. but i could be wrong. 

see this: https://learn.adafruit.com/adafruit-data-logger-shield/using-the-real-time-clock

**Comment by walkerjeffd** [comment]
ah, you beat me to it. but yeah, the cheaper one might do just as well. only way to find out probably would be to run a test under changing temperature conditions (over a month or two) and see if the two RTCs stay in sync or not.

**Comment by dwblair** [comment]
Yes, great point -- and what would be acceptably 'in sync' for a typical application in hydrology -- any idea?  E.g. I just saw a blog post  http://limchonghan.wordpress.com/ that claims losing a few seconds per day in relatively stable room temperature conditions.  
- Do events across sensors need to be correlated, one with the other, in time?  With what accuracy / resolution?  I guess it'll depend on the application ... but if e.g. there are many applications in which it doesn't matter so much, then maybe we don't need an RTC for many of those applications -- e.g. noticing conductivity spiking relative to baseline, and pinging an internet connection, wouldn't really need an RTC on board at all, as all of the packets sent would be timestamped on the server ... but correlating conductivity spikes with storm pulses in a river network might require a certain level of on-board timing accuracy ...

**Comment by walkerjeffd** [comment]
I think we need an RTC for everything in order to get the timestamp (whenever I say timestamp, i refer to the date and time, e.g. 2014-05-30T13:58). Trying to figure out the timestamp from recorded millis() times would be a major pain (would have to know the time when the device powered on, and would be impossible if it loses power during a deployment and wakes up again).

Say it looses (or gains) 10 sec/day. THen over 30 days, that would be 10*30/60 = 5 minutes. So we'd gain or loose 5 minutes per month. I think that's ok. But it might be a much larger change when sitting in the environment (not only air temperature changing +/- 20 degrees per day, but the effect of solar radiation might heat the thing up even more if its on the surface). So probably need a test.

What exactly are the differences between a cheap and expensive RTC in terms of cost, power consumption, programming? Would it be possible to easily switch them without having to change the code much?

**Comment by dwblair** [comment]
Oh, and in any case, now that the boards have arrived, I realize: we don't need a separate battery for the RTC. ( I'd forgotten -- Ben had already pointed this out when designing the Riffle).  The RTC battery is typically to keep track of time when the main power to the device goes off; but if that happens for us, all is lost anyway :) So we'll just run the RTC off VCC.  Which is cool -- it'll save board space and cost.

**Comment by walkerjeffd** [comment]
Well the RTC battery is also to keep the clock running between deployments. If there is no battery, then how will the RTC know what time it is when it is first turned on? You'd have to manually set the time on the RTC at the start of each deployment. Or if it runs out of power in the field and you need to just change the AA batteries, then you'd still have to connect it∂ to a computer to set the timestamp. If there is an RTC battery, then you'd only have to set the timestamp every few months or maybe once per year depending on how much drift there is in the RTC (with the temp-compensated RTC, you may be able to just set it once per 7-8 years).

**Comment by dwblair** [comment]
I'm so glad that I've got external human minds on this problem :)  Thanks -- you just recapitulated the discussion we'd had a few weeks ago about this; we put the RTC battery there for good reason!  And nice to have it written up in the comments here.  Cheers!
